server/protocols/protocols-library/src/main/java/org/apache/james/protocols/lib/ProtocolHandlerChainImpl.java

```diff
@@ -158,9 +158,6 @@ public <T> Optional<T> getFirstHandler(Class<T> type) {
      */
     @Override
     public void destroy() {
-        LinkedList<ProtocolHandler> lHandlers = getHandlers(ProtocolHandler.class);
-        for (ProtocolHandler handler : lHandlers) {
-            handler.destroy();
-        }
+        getHandlers(ProtocolHandler.class).stream().parallel().forEach(ProtocolHandler::destroy);

```

Comment:
Is this making a difference? I would expect those handlers to not be linked to netty...

Reply:
Yes please. Handlers shall be quick to dispose. (+ I'm opinionated on parallel streams... - sorry)